internal/domain/file_content_response.go

```diff
@@ -0,0 +1,9 @@
+package domain
+
+type FileContentResponse struct {
+	Filename  string `json:"filename"`
+	Content   string `json:"content"`
```

Comment:
If `Position` is meant to be a zero-based index, clarify its usage in the code. If not, ensure that it is used consistently across your application to avoid confusion.